src/components/ConfirmRequestDialog.test.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import ConfirmRequestDialog from './ConfirmRequestDialog';
+import DocumentTypesFixture from '../../cypress/fixtures/document-types-response.json';
+import { ResponseMapper } from '../boundary/response-mapper';
+import { EvidenceRequestRequest } from '../gateways/internal-api';
+
+const onAccept = jest.fn();
+const onDismiss = jest.fn();
+const DocumentTypes = DocumentTypesFixture.map(ResponseMapper.mapDocumentType);
+
+describe('Confirm Request Dialog', () => {
+  describe('when there is no request', () => {
+    beforeEach(() => {
+      render(
+        <ConfirmRequestDialog
+          onDismiss={onDismiss}
+          onAccept={onAccept}
+          documentTypes={DocumentTypes}
+        />
+      );
+    });
+
+    it('is closed', () => {
+      expect(screen.queryByRole('dialog')).toBeNull();
+    });
+  });
+
+  describe('when there is a request', () => {
+    const baseRequest: EvidenceRequestRequest = {
+      documentTypes: ['passport'],
+      deliveryMethods: [],
+      resident: {
+        name: 'Frodo',
+        email: 'frodo@bagend.com',
+        phoneNumber: '0123',
+      },
+    };
+
+    beforeEach(() => {
+      render(
+        <ConfirmRequestDialog
+          request={baseRequest}
+          onDismiss={onDismiss}
+          onAccept={onAccept}
+          documentTypes={DocumentTypes}
+        />
+      );
+    });
+
+    it('is closed', () => {
+      expect(screen.queryByRole('dialog')).toBeVisible();
+    });
+
+    it('has the correct text', () => {
+      expect(
+        screen.getByText("You're about to make a request to:")
+      ).toBeVisible();
+
+      Object.values(baseRequest.resident).forEach((text) =>
+        expect(screen.getByText(text)).toBeVisible()
+      );
+
+      expect.assertions(4);
```

Comment:
Shouldn't we check for the exact words in the text here?

Reply:
We are—we show a list where each bullet point is literally just the attribute of the resident, so this should be enough 
![image](https://user-images.githubusercontent.com/6242344/105981354-e2ffc400-608d-11eb-9858-22291bd4f841.png)